Tests/ML/pipelines/test_forward_pass.py

```diff
@@ -158,13 +164,22 @@ def test_amp_activated(use_model_parallel: bool,
         else:
             raise ValueError(f"Expected this call to succeed, but got: {ex}")
 
-    # This is the same logic spelt out in update_model_for_multiple_gpu
+    model_and_info.try_create_optimizer_and_load_from_checkpoint()
+
+    model = model_and_info.model
+    optimizer = model_and_info.optimizer
+
+    # for mypy
+    assert model is not None
+    assert optimizer is not None
+
```

Comment:
Instead of having those checks, you could have a getter on ModelAndInfo, that raises if the model is not yet created. i see no case where a None model would be a valid outcome.